Use webjars and jsdeps.js

PerWiklander opened the pull request:
The app runs but the tests fail with this, you might want to take a stab at it, probably some test related settings I haven't figured out yet:

```
java.lang.RuntimeException: Exception while running JS code: TypeError: Cannot read property "defaultView" from undefined (/Users/per/.ivy2/cache/org.webjars/jquery/jars/jquery-1.11.1.jar#META-INF/resources/webjars/jquery/1.11.1/jquery.js#1049)
    at scala.sys.package$.error(package.scala:27)
    at org.scalajs.jsenv.rhino.RhinoJSEnv.org$scalajs$jsenv$rhino$RhinoJSEnv$$internalRunJS(RhinoJSEnv.scala:168)
    at org.scalajs.jsenv.rhino.RhinoJSEnv$Runner.run(RhinoJSEnv.scala:60)
    at org.scalajs.sbtplugin.FrameworkDetector.detect(FrameworkDetector.scala:64)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$40.apply(ScalaJSPluginInternal.scala:453)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$40.apply(ScalaJSPluginInternal.scala:438)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:235)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[error] (sharedProjectJS/test:loadedTestFrameworks) Exception while running JS code: TypeError: Cannot read property "defaultView" from undefined (/Users/per/.ivy2/cache/org.webjars/jquery/jars/jquery-1.11.1.jar#META-INF/resources/webjars/jquery/1.11.1/jquery.js#1049)
[error] Total time: 15 s, completed 2015-feb-23 02:30:29
```

PerWiklander committed:
Use webjars and jsdeps.js

PerWiklander commented:
I think this is the missing part:

jsDependencies += RuntimeDOM

according to the official tutorial here: http://www.scala-js.org/doc/tutorial.html
I don't have time to try it now though.

ochrons commented:
Can you make the necessary modification to your PR to add that `jsDependencies += RuntimeDOM`

PerWiklander committed:
Add runtime dom as js dependency so that we can run tests from the command line.

The underlying reason for this is that jQuery is now a js dependency and will be included and fail to find a dom.

PerWiklander commented:
The tests now pass, on my workstation.

The docs mention

> If you use the fastOpt stage, this will switch to running your code with PhantomJS instead of Node.js, which you need to install. Otherwise, in Rhino, a fake DOM is automatically made available.

So I'm not 100% sure how to interpret that and if this will work out of the box everywhere.

PerWiklander committed:
We only need the runtime dom for tests
